src/Home.jsx

```diff
@@ -23,13 +55,31 @@ class Home extends React.Component {
     this.setState({ toggledBar: !toggledBar });
   }
 
+  async callArea(selectedElement) {
+    const { sedimentaryList } = this.state;
+    try {
+      const response = await RestAPI.requestAreaSelected(selectedElement.name);
+      this.setState({
+        selectedArea: {
+          sedimentary_name: sedimentaryList.filter((item) => (item.code === response.sedimentary_code))[0].name,
```

Comment:
Since you are looking for the first match to `response.sedimentary_code` you should use `find` instead of filter, so you get a value and not an array, and you don't go through all the list

Reply:
Thank you. Solved.